src/app/Http/Middleware/CheckChoiceRole.php

```diff
@@ -16,9 +16,9 @@ class CheckChoiceRole
      * @param  \Closure  $next
      * @return mixed
      */
-    public function handle($request, Closure $next, ...$roles)
+    public function handle($request, Closure $next, $roles)
     {
-        if (!check_role(Auth::user()->presentations()->where('id', $request->choice->poll->presentation->id)->first(), $roles)) {
+        if (!check_role(Auth::user()->presentations()->where('id', $request->choice->poll->presentation->id)->first(), explode('|', $roles))) {
```

Comment:
`...$roles` like in java is to be able to do a call with variable nb of params ex `myFunction(role1, role2, role3, etc.)`
also Laravel conventions for calling function with params c.f validators are `myFunction:param1,param2|myOtherFunction` so you don't want to force it to use a `|` to séparate your params as it means by Laravel that you are waiting another function.
Not sure if it's understandable, don't hesitate to ask questions

Reply:
I don't want multiple variables, I want an array, otherwise the helper won't work. I didn't find any other way to pass an array to the middleware